### PR TITLE
Use str(Label(...)) in tensorflow.bzl for use as a submodule

### LINKED ISSUE #8439: tf_kernel_library doesn't work from repo containing tensorflow as a submodule
`tf_kernel_library` and friends from `tensorflow.bzl` refer to deps like `//tensorflow/core:gpu_lib`.  These deps do not work if TensorFlow exists only under `@org_tensorflow`, which results in errors like

    bazel build ...
    WARNING: /usr/local/google/home/geoffreyi/.cache/bazel/_bazel_geoffreyi/f928d2088d4c5ef4e2f54ff2af6f8a71/external/org_tensorflow/tensorflow/workspace.bzl:72:5: tf_repo_name was specified to tf_workspace but is no longer used and will be removed in the future.
    WARNING: /usr/local/google/home/geoffreyi/.cache/bazel/_bazel_geoffreyi/f928d2088d4c5ef4e2f54ff2af6f8a71/external/org_fold/WORKSPACE:1: Workspace name in /usr/local/google/home/geoffreyi/.cache/bazel/_bazel_geoffreyi/f928d2088d4c5ef4e2f54ff2af6f8a71/external/org_fold/WORKSPACE (@org_tensorflow_fold) does not match the name given in the repository's definition (@org_fold); this will cause a build error in future versions.
    ERROR: /usr/local/google/home/geoffreyi/deepmath/deepmath/guidance/BUILD:83:1: no such package 'tensorflow': Package crosses into repository @org_tensorflow and referenced by '//deepmath/guidance:clause_ops'.
    ERROR: /usr/local/google/home/geoffreyi/deepmath/deepmath/guidance/BUILD:83:1: no such package 'tensorflow': Package crosses into repository @org_tensorflow and referenced by '//deepmath/guidance:clause_ops'.
    ERROR: /usr/local/google/home/geoffreyi/deepmath/deepmath/guidance/BUILD:83:1: no such package 'tensorflow': Package crosses into repository @org_tensorflow and referenced by '//deepmath/guidance:clause_ops'.
    ERROR: /usr/local/google/home/geoffreyi/deepmath/deepmath/guidance/BUILD:83:1: no such package 'tensorflow': Package crosses into repository @org_tensorflow and referenced by '//deepmath/guidance:clause_ops'.
    ERROR: /usr/local/google/home/geoffreyi/deepmath/deepmath/guidance/BUILD:83:1: no such package 'tensorflow': Package crosses into repository @org_tensorflow and referenced by '//deepmath/guidance:clause_ops'.
    ERROR: /usr/local/google/home/geoffreyi/deepmath/deepmath/guidance/BUILD:83:1: no such package 'tensorflow': Package crosses into repository @org_tensorflow and referenced by '//deepmath/guidance:clause_ops'.
    ERROR: Analysis of target '//deepmath/guidance:clause_ops' failed; build aborted.
    INFO: Elapsed time: 0.320s

### PR DESCRIPTION
**Pull request by girving**
tf_kernel_library and friends do not currently work if called from a
repo that contains tensorflow as submodule.  Justine recommended this
as the right fix.

Fixes #8439.

### CODE REVIEW

**Comment by martinwicke** [comment]
Jenkins, test this please (the makefile failure may have been due to a recent reactivation of a broken slave)

**Comment by gunan** [comment]
We just upgraded to bazel 0.4.5.
@jart With 0.4.5 do we have a better solution we can use now? maybe just editing these rules to be `@org_tensorflow://....` ?

**Comment by girving** [comment]
@gunan @jart Can you decide whether this PR or https://github.com/tensorflow/tensorflow/pull/8763 is the better one?

**Comment by gunan** [comment]
I think the other PR is much better. (If we can get it working, there seems to be a test issue with it.)

**Comment by gunan** [comment]
The other solution is not working even with bazel 0.4.5, so we will go with this solution for a little while longer.

**Review by gunan — APPROVED** [review]

**Comment by girving** [comment]
@gunan Thanks!

**Comment by girving** [comment]
This PR broke inside Google, so I'm going to redo it from the inside.

File: tensorflow/tensorflow.bzl [hunk 1]
```diff
@@ -27,79 +27,79 @@ load(
 
 # List of proto files for android builds
 def tf_android_core_proto_sources(core_proto_sources_relative):
-  return ["//tensorflow/core:" + p
+  return [str(Label("//tensorflow/core:" + p))
```
Comment (hawkinsp) [inline tensorflow/tensorflow.bzl]: Could you use the clean_dep() wrapper defined here?
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tensorflow.bzl#L49

I think it's a bit more readable. Same everywhere.

**Pull request merged**